sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/address/ParameterableImpl.java

```diff
@@ -155,13 +159,44 @@ public void setParameter(String name, String value) {
 		this.parameters.put(name.toLowerCase(), value);
 		if(header != null) {
 			try {
-				header.setParameter(name, "".equals(value) ? null : value);
+				if (isQuotableParam(name) && (header instanceof ParametersExt)){
+					((ParametersExt) header).setQuotedParameter(name, value);
+				}else{
+					header.setParameter(name, "".equals(value) ? null : value);
+				}
 			} catch (ParseException e) {
 				throw new IllegalArgumentException("Problem setting parameter",e);
 			}
 		}
 	}
-
+	
+	/*
+	 * Check if the param name is in the list that its value need to be quoted
+	 * 
+	 * @param name - a string specifying the parameter name
+	 * 
+	 */
+	private boolean isQuotableParam(String name){
+		boolean isQuotableParameter = false;
+		SipContext context = SipApplicationSessionCreationThreadLocal.lookupContext();
+		if (context != null){
+			Object object = context.getServletContext().getAttribute("org.restcomm.servlets.sip.QUOTABLE_PARAMETER");

```

Comment:
could we build a default value for this attribute, including known attributes from SIP RFCs?